### PR TITLE
feat: add guest nickname handling

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/17147
Fix https://github.com/nextcloud-libraries/nextcloud-browser-storage/issues/3

Followup from https://github.com/nextcloud-libraries/nextcloud-sharing/pull/33

### CODE REVIEW

File: lib/guest.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { getBuilder } from '@nextcloud/browser-storage'
```
Comment (susnux) [inline lib/guest.ts:1]: Need to add to dependencies

**Review by ShGKme — APPROVED** [review]

**Comment by ShGKme** [comment]
Thanks for moving it here 💙

File: lib/guest.ts (hunk 1 above)
Comment (ShGKme) [inline lib/guest.ts:1]:
```suggestion
/**
 * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

import { getBuilder } from '@nextcloud/browser-storage'
```

File: lib/guest.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const browserStorage = getBuilder('public').persist().build()
+
+/**
+ * Get the guest nickname for public shares
+ */
+export function getGuestNickname(): string | null {
```
Comment (ChristophWurst) [inline lib/guest.ts:8]: Useful for nickvergessen

**Review by ChristophWurst — APPROVED** [review]

File: lib/guest.ts (hunk 2 above)
Comment (ShGKme) [reply lib/guest.ts:8]: Is it supposed to be used for only and only public shares?

Comment (skjnldsv) [reply lib/guest.ts:8]: @ShGKme the definition of public share can be argued :see_no_evil: 
We currently use if tor public links. E.g Talk, Files, Polls, Text, Colabora....

Comment (ShGKme) [reply lib/guest.ts:12]: But not for, e.g., Talk guest user?

Comment (susnux) [reply lib/guest.ts:12]: Maybe just reword to `public pages` as a public share is a public page :sweat_smile:

**Review by susnux — APPROVED** [review]

File: lib/guest.ts (hunk 2 above)
Comment (skjnldsv) [reply lib/guest.ts:12]: > But not for, e.g., Talk guest user?

hmm, yes? I'll reword anyway :see_no_evil:

**Pull request merged**